spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jProperties.java

```diff
@@ -142,4 +156,55 @@ public void setEnabled(boolean enabled) {
 
 	}
 
+	public static class Indexes {
+
+		private AutoIndexMode auto = AutoIndexMode.NONE;
+
+		private Dump dump = new Dump();
+
+		public AutoIndexMode getAuto() {
+			return this.auto;
+		}
+
+		public void setAuto(AutoIndexMode auto) {
+			this.auto = auto;
+		}
+
+		public Dump getDump() {
+			return this.dump;
+		}
+
+		public void setDump(Dump dump) {
+			this.dump = dump;
+		}
+
+		public static class Dump {
+
+			/**
+			 * Generated Indexes Output Dir.
+			 */
+			private String dir = ".";
+
+			/**
+			 * Generated Indexes Output Filename.
+			 */
+			private String filename = "generated_indexes.cql";
```

Comment:
How often are you going to use that feature? I am asking because it complexifies the configuration setup quite a lot.